src/components/tooltip/use-tooltip-positioning.test.js

```diff
@@ -0,0 +1,124 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import useTooltipPositioning from './use-tooltip-positioning.js';
+
+describe('use-tooltip-positioning', () => {
+  const requiredProps = {
+    triangleHeight: 6,
+    tooltipRef: { current: { getBoundingClientRect: () => ({}) } },
+  };
+
+  it('visible is false initially', () => {
+    const { result } = renderHook(() => useTooltipPositioning(requiredProps));
+    expect(result.current.visible).toBe(false);
+  });
+
+  it('visible is true after show is called', () => {
+    const target = { getBoundingClientRect: () => ({}) };
+
+    const { result } = renderHook(() => useTooltipPositioning(requiredProps));
+    act(() => result.current.show({ target }));
+    expect(result.current.visible).toBe(true);
+  });
+
+  it('is not visible after hide is called', () => {
+    const target = { getBoundingClientRect: () => ({}) };
+
+    const { result } = renderHook(() => useTooltipPositioning(requiredProps));
+    act(() => result.current.show({ target }));
+    act(() => result.current.hide());
+    expect(result.current.visible).toBe(false);
+  });
+
+  describe('position and direction', () => {
+    const tooltipRect = {
+      height: 20,
+      width: 40,
+      x: 60,
+      y: 80,
+    };
+    const triangleHeight = 5;
+    const sourceRect = {
+      height: 2,
+      width: 4,
+      x: 6,
+      y: 8,
+    };
+    const target = { getBoundingClientRect: () => (sourceRect) };
+
+    it('direction = top sets the position central horizontally and above vertically', () => {
+      const { result } = renderHook(() => useTooltipPositioning({
+        direction: 'top',
+        triangleHeight,
+        tooltipRef: {
+          current: {
+            getBoundingClientRect: () => tooltipRect,
+          },
+        },
+      }));
+
+      act(() => result.current.show({ target }));
+
+      expect(result.current.position).toEqual({
+        left: sourceRect.x + (sourceRect.width / 2) - (tooltipRect.width / 2),
+        top: sourceRect.y - tooltipRect.height - triangleHeight,
+      });
```

Comment:
I didn't like using the actual calculations here, but I was worried about using hardcoded numbers as being unclear to future engineers. I can add them with comments though if that's preferred to the arithmetic